### PR TITLE
Fix typing bug and make Readme clearer

### LINKED ISSUE #686: tslint.d.ts exported member does not match JavaScript code
Currently the `tslint.d.ts` file contains the following:
```js
declare module "tslint" {
    export = Lint;
}
```

but the code used to invoke the linter is:
```js
var Linter = require("tslint");
var ll = new Linter(fileName, contents, options);
```

For this to work the .d.ts file has to be changed to:
```js
declare module "tslint" {
    export = Lint.Linter;
}
```

### PR DESCRIPTION
**Pull request by JKillian**
Fixes #686

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -245,10 +245,11 @@ TSLint ships with a set of core rules that can be configured. However, users are
 
 Rule names are always camel-cased and *must* contain the suffix `Rule`. Let us take the example of how to write a new rule to forbid all import statements (you know, *for science*). Let us name the rule file `noImportsRule.ts`. Rules can be referenced in `tslint.json` in their dasherized forms, so `"no-imports": true` would turn on the rule.
 
-Now, let us first write the rule in TypeScript. At the top, we reference TSLint's [definition](https://github.com/palantir/tslint/blob/master/lib/tslint.d.ts) file. The exported class name must always be named `Rule` and extend from `Lint.Rules.AbstractRule`.
+Now, let us first write the rule in TypeScript. At the top, we reference TSLint's [definition](https://github.com/palantir/tslint/blob/master/lib/tslint.d.ts) file and the [definition file](https://github.com/palantir/tslint/blob/master/typings/typescriptServices.d.ts) for TypeScript's services. The exported class name must always be named `Rule` and extend from `Lint.Rules.AbstractRule`.
 
 ```typescript
-/// <reference path='tslint.d.ts' />
+/// <reference path='lib/tslint.d.ts' />
+/// <reference path='typings/typescriptServices.d.ts' />
```
Comment (adidahiya) [inline README.md]: does ordering matter here? perhaps typescriptServices should come first

Comment (JKillian) [inline README.md]: Hmm, seems like it might. I'll switch those around

**Comment by JKillian** [comment]
[Changes made]

File: README.md [hunk 2]
```diff
@@ -245,10 +245,11 @@ TSLint ships with a set of core rules that can be configured. However, users are
 
 Rule names are always camel-cased and *must* contain the suffix `Rule`. Let us take the example of how to write a new rule to forbid all import statements (you know, *for science*). Let us name the rule file `noImportsRule.ts`. Rules can be referenced in `tslint.json` in their dasherized forms, so `"no-imports": true` would turn on the rule.
 
-Now, let us first write the rule in TypeScript. At the top, we reference TSLint's [definition](https://github.com/palantir/tslint/blob/master/lib/tslint.d.ts) file. The exported class name must always be named `Rule` and extend from `Lint.Rules.AbstractRule`.
+Now, let us first write the rule in TypeScript. At the top, we reference TSLint's [definition](https://github.com/palantir/tslint/blob/master/lib/tslint.d.ts) file and the [definition file](https://github.com/palantir/tslint/blob/master/typings/typescriptServices.d.ts) for TypeScript's services. The exported class name must always be named `Rule` and extend from `Lint.Rules.AbstractRule`.
```
Comment (adidahiya) [inline README.md]: some nitpicks:

- make the first link (to tslint.d.ts) also apply to the word "file" (so both links are "definition file")
- TypeScript's services -> TypeScript's language services
- "class name must always be named" -> "class must always be named"

**Comment by JKillian** [comment]
[changes made]

**Pull request merged**